dive_core/tests/perf_metrics_data_test.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#include "dive_core/perf_metrics_data.h"
+#include "dive_core/available_metrics.h"
+#include "gtest/gtest.h"
+#include <filesystem>
+#include "gmock/gmock.h"
+
+namespace Dive
+{
+
+namespace
+{
+
+MATCHER_P(PerfMetricsRecordEq, expected, "has the correct perf metrics record fields")
+{
+    EXPECT_EQ(arg.m_context_id, expected.m_context_id);
+    EXPECT_EQ(arg.m_process_id, expected.m_process_id);
+    EXPECT_EQ(arg.m_frame_id, expected.m_frame_id);
+    EXPECT_EQ(arg.m_cmd_buffer_id, expected.m_cmd_buffer_id);
+    EXPECT_EQ(arg.m_draw_id, expected.m_draw_id);
+    EXPECT_EQ(arg.m_draw_label, expected.m_draw_label);
+    EXPECT_EQ(arg.m_program_id, expected.m_program_id);
+    EXPECT_EQ(arg.m_draw_type, expected.m_draw_type);
+    EXPECT_EQ(arg.m_lrz_state, expected.m_lrz_state);
+    return true;
+}
+
+class PerfMetricsDataTest : public ::testing::Test
+{
+protected:
+    void SetUp() override
+    {
+        m_available_metrics = Dive::AvailableMetrics::LoadFromCsv(TEST_DATA_DIR
+                                                                  "/mock_available_metrics.csv");
+        ASSERT_TRUE(m_available_metrics);
+    }
+    std::optional<Dive::AvailableMetrics> m_available_metrics;
+};
+
+TEST_F(PerfMetricsDataTest, LoadFromCsv)
+{
+    auto perf_metrics_data = Dive::PerfMetricsData::LoadFromCsv(TEST_DATA_DIR
+                                                                "/mock_perf_metrics_data.csv",
+                                                                *m_available_metrics);
+    ASSERT_TRUE(perf_metrics_data);
```

Comment:
This will work, but I was confused whether perf_metrics_data was a bool. For readability, I'd suggest using a comparison with nullptr.

There's a couple of ways to do this. This first suggestion is my preferred suggestion since it reads a little more like an English phrase:
```suggestion
using ::testing::Not;
using ::testing::IsNull();
    ASSERT_THAT(perf_metrics_data, Not(IsNull()));
```
This changes the output like so:
```c++
ASSERT_TRUE(perf_metrics_data);
// Value of: perf_metrics_data
//  Actual: false
// Expected: true

ASSERT_THAT(perf_metrics_data, Not(IsNull()));
// Value of: perf_metrics_data
// Expected: isn't NULL
//   Actual: (nullptr)
```

Alternatively, this is also an option
```suggestion
    ASSERT_NE(perf_metrics_data, nullptr);
```